frontend/lib/models/test_results.dart

```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/material.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:yaru/yaru.dart';
+import 'package:yaru_icons/yaru_icons.dart';
+
+part 'test_results.freezed.dart';
+part 'test_results.g.dart';
+
+@freezed
+class TestResult with _$TestResult {
+  const factory TestResult({
+    required int id,
+    required String name,
+    required String type,
+    required TestResultStatus status,
```

Comment:
Nice, I see you used an enum for test result status (I guess we could use one in the backend too). How about `type` can it be an enum?

Reply:
I firstly thought that this `type` field has some meaning, but I then noticed that it always has the value `test`. I think it would be best to remove it as it doesn't seem to have any meaning behind it and we don't use it in any of our views.